sway-core/src/convert_parse_tree.rs

```diff
@@ -2863,11 +2863,36 @@ fn assignable_to_expression(
             index: Box::new(expr_to_expression(ec, *arg.into_inner())?),
             span,
         },
-        Assignable::FieldProjection { target, name, .. } => Expression::SubfieldExpression {
-            prefix: Box::new(assignable_to_expression(ec, *target)?),
-            field_to_access: name,
-            span,
-        },
+        Assignable::FieldProjection { target, name, .. } => {
+            let mut idents = vec![&name];
+            let mut base = &*target;
+            let storage_access_field_names_opt = loop {
+                match base {
+                    Assignable::FieldProjection { target, name, .. } => {
+                        idents.push(name);
+                        base = target;
+                    }
+                    Assignable::Var(name) => {
+                        if name.as_str() == "storage" {
```

Comment:
Man, I was looking at this code trying to work out how it was related to storage stuff, and here it is!  Feels a bit hacky matching against a string like this -- perhaps the parser should become more storage-aware?

Reply:
Absolutely. Let me create an issue. 

I think Andrew probably wanted to commonize storage accesses and subfield accesses as much as possible.

Reply:
https://github.com/FuelLabs/sway/issues/1986